ADA/databricks_fundamentals.qmd

```diff
@@ -0,0 +1,232 @@
+---
+title: "Databricks fundamentals"
+---
+
+------------------------------------------------------------------------
+
+# What is Databricks?
+
+------------------------------------------------------------------------
+
+Databricks is a web based platform for large scale data manipulation and analysis using code to create reproducible data pipelines. Primarily it takes the form of a website which you can create data pipelines and perform analysis in. It currently supports the languages R, SQL, python and scala, and integrates well with Git based version control systems such as GitHub or Azure DevOps.
+
+Behind the scenes it is a distributed cloud computing platform which utilizes the [Apache Spark engine](https://spark.apache.org/) to split up heavy data processing into smaller chunks. It then distributes them to different 'computers' within the cloud to perform the processing of each chunk in parallel. Once each 'computer' is finished processing the results are recombined and passed back to the user or stored.
+
+Due to the parallel processing capabilities this improves the performance of the data processing and allows for the manipulation of very large data sets in a relatively short amount of time.
+
+In addition, it also provides new tools within the platform to construct and automate complex data transformations and processing.
+
+------------------------------------------------------------------------
+
+## Key differences
+
+Underpinning the technology are some key differences in how computers we're familiar with, and Databricks (and distributed computing in general) are structured.
+
+### Traditional computing
+
+Currently, we are used to using a PC or laptop to do our data processing. A traditional computer has all of the components it needs to function:
+
+-   A processor and memory to do calculations
+-   A hard drive to store data permanently on
+-   A keyboard and mouse to capture user input
+-   A screen to provide outputs to the user
+
+![](/images/ada-traditional-computer.jpg){width="273"}
+
+Any traditional computer is limited by it's hardware meaning there is an upper limit on the size and complexity of data it can process.
+
+In order to increase the amount of data a computer can process you would have to switch out the physical hardware of the machine for something more powerful.
+
+### On Databricks
+
+In Databricks you can scale the components of your machine up (CPU cores, RAM) without having to build a physical machine to house them, essentially temporarily 'borrowing' processor power, memory and storage from a super computer.
+
+This means you can perform very heavy analyses that your laptop wouldn't be able to cope with. The Databricks platform takes the place of the keyboard/mouse and screen, taking input and providing output back to the user.
```

Comment:
> The Databricks platform takes the place of the keyboard/mouse and screen, taking input and providing output back to the user.

This sounds a little bit like the user has less control of the input / that Databricks is a "self driving machine", could we rephrase? I see what you're getting at though

Reply:
Have re-phrased in the latest commit to:

"This means you can perform very heavy analyses that your laptop wouldn't be able to cope with. The Databricks platform provides a way for you to interface with the cloud computer in place of the keyboard/mouse and screen, taking your inputs and providing the resulting outputs back to you."